src/matrix/storage/idb/StorageFactory.js

```diff
@@ -21,14 +21,18 @@ import { schema } from "./schema.js";
 import { detectWebkitEarlyCloseTxnBug } from "./quirks.js";
 
 const sessionName = sessionId => `hydrogen_session_${sessionId}`;
-const openDatabaseWithSessionId = sessionId => openDatabase(sessionName(sessionId), createStores, schema.length);
+const openDatabaseWithSessionId = function(sessionId, idbFactory) {
+    return openDatabase(sessionName(sessionId), createStores, schema.length, idbFactory);
+}
 
 async function requestPersistedStorage() {
-    if (navigator?.storage?.persist) {
-        return await navigator.storage.persist();
-    } else if (document.requestStorageAccess) {
+    // don't assume browser so we can run in node with fake-idb
+    const glob = this;
```

Comment:
This doesn't work in strict mode which the SDK `hydrogen.cjs.js` has at the top `"use strict";`.